extensions/objects/templates/svg-filename-editor.component.ts

```diff
@@ -598,6 +600,97 @@ angular.module('oppia').component('svgFilenameEditor', {
+        var size = ctrl.fabricjsOptions.size;
+        var circle = new fabric.Circle({
+          left: left,
+          top: top,
+          radius: parseInt(size.substring(0, size.length - 2)) + 2,
+          fill: '#666666',
+          stroke: '#666666',
+          hasBorders: false,
+          hasControls: false
+        });
+        return circle;
+      };
+
+      var drawQuadratic = function() {
+        var size = ctrl.fabricjsOptions.size;
+        var curve = new fabric.Path('M 40 40 Q 95, 100, 150, 40', {
+          stroke: ctrl.fabricjsOptions.stroke,
+          fill: ctrl.fabricjsOptions.fill,
+          strokeWidth: parseInt(size.substring(0, size.length - 2)),
+          objectCaching: false,
+          selectable: false
+        });
+        ctrl.canvas.add(curve);
+
+        var p1 = makeCurveCircle(95, 100);
+        p1.name = 'p1';
+        p1.set({
+          radius: 12,
+          fill: '#ffffff',
+          strokeWidth: 5
+        });
+        ctrl.canvas.add(p1);
+
+        var p0 = makeCurveCircle(40, 40);
+        p0.name = 'p0';
+        ctrl.canvas.add(p0);
+
+        var p2 = makeCurveCircle(150, 40);
+        p2.name = 'p2';
+        ctrl.canvas.add(p2);
+      };
+
+      ctrl.createBezier = function() {
+        if (ctrl.drawMode === DRAW_MODE_NONE) {
+          ctrl.canvas.discardActiveObject();
+          ctrl.drawMode = DRAW_MODE_BEZIER;
+          ctrl.canvas.getObjects().forEach(function(item) {
+            item.set({
+              hoverCursor: 'default',
+              selectable: false
+            });
+          });
+          drawQuadratic();
+        } else {
+          // Remove the circles and the current path.
+          ctrl.canvas.getObjects().slice(-3).forEach(function(item) {
+            ctrl.canvas.remove(item);
+          });
+          var path = ctrl.canvas.getObjects().slice(-1)[0].get('path');
+          ctrl.canvas.remove(ctrl.canvas.getObjects().slice(-1)[0]);
+          ctrl.canvas.getObjects().forEach(function(item) {
+            item.set({
+              hoverCursor: 'move',
+              selectable: true
+            });
+          });
+          // Change mode and then add the path so that the object is added in
+          // cavasObjects array.
+          ctrl.drawMode = DRAW_MODE_NONE;
+          // Adding a new path so that the bbox is computed correctly.
+          var size = ctrl.fabricjsOptions.size;
+          var curve = new fabric.Path(path, {
+            stroke: ctrl.fabricjsOptions.stroke,
+            fill: ctrl.fabricjsOptions.fill,
+            strokeWidth: parseInt(size.substring(0, size.length - 2)),
+          });
+          ctrl.canvas.add(curve);
+        }
+      };
+
+      ctrl.isBezierEnabled = function() {
+        return Boolean(
+          ctrl.areAllToolsEnabled() ||
+          ctrl.drawMode === DRAW_MODE_BEZIER);
+      };
+
+      ctrl.isDrawModeBezier = function() {
+        return Boolean(ctrl.drawMode === DRAW_MODE_BEZIER);
```

Comment:
Why type cast here?

Reply:
I have removed the typecast. No particular reason for doing this way. Done.